tuna/miopen/import_configs.py

```diff
@@ -156,87 +99,83 @@ def insert_config(driver, counts, dbt, args):
 
   return new_cf.id
 
 
-def process_config_line_v2(driver, args, counts, dbt):
+def process_config_line_v2(driver, args, counts, dbt, logger):
   """Assumes config passed already exists and will skip the insert step
         if tag_only present. Otherwise it will first try and insert and
         then tag."""
-  if args.tag_only:
-    _ = tag_config_v2(driver, counts, dbt, args, new_cf=None)
+  if args.import_configs.tag_only:
+    _ = tag_config_v2(driver, counts, dbt, args, logger, new_cf=None)
     return False
 
-  _ = insert_config(driver, counts, dbt, args)
+  _ = insert_config(driver, counts, dbt, args, logger)
   return True
 
 
-def parse_line(args, line, counts, dbt):
+def parse_line(args, line, counts, dbt, logger):
   """parse a driver line or fdb line from an input file and insert the config"""
   if args.config_type == ConfigType.batch_norm:
-    driver = DriverBatchNorm(line, args.command)
+    driver = DriverBatchNorm(line, args.import_configs.command)
   else:
-    driver = DriverConvolution(line, args.command)
+    driver = DriverConvolution(line, args.import_configs.command)
 
-  if not args.batch_list:
-    process_config_line_v2(driver, args, counts, dbt)
+  if not args.import_configs.batch_list:
+    process_config_line_v2(driver, args, counts, dbt, logger)
   else:
-    for bsz in args.batch_list:
-      LOGGER.info('Batchsize: %s', bsz)
+    for bsz in args.import_configs.batch_list:
+      logger.info('Batchsize: %s', bsz)
       driver.batchsize = bsz
-      process_config_line_v2(driver, args, counts, dbt)
+      process_config_line_v2(driver, args, counts, dbt, logger)
 
   return True
 
 
-def import_cfgs(args, dbt):
+def import_cfgs(args, dbt, logger):
   """import configs to mysql from file with driver invocations"""
   connect_db()
 
   counts = {}
   counts['cnt_configs'] = 0
   counts['cnt_tagged_configs'] = set()
-  with open(os.path.expanduser(args.file_name), "r") as infile:  # pylint: disable=unspecified-encoding
+  with open(os.path.expanduser(args.import_configs.file_name), "r") as infile:  # pylint: disable=unspecified-encoding
     for line in infile:
       try:
-        parse_line(args, line, counts, dbt)
+        parse_line(args, line, counts, dbt, logger)
       except ValueError as err:
-        LOGGER.warning(err)
+        logger.warning(err)
         continue
 
   return counts
 
 
-def main():
+def run_import_configs(args, logger):
   """Main function"""
-  args = parse_args()
   counts = {}
 
   dbt = MIOpenDBTables(session_id=None, config_type=args.config_type)
 
-  counts = import_cfgs(args, dbt)
-
-  LOGGER.info('New configs added: %u', counts['cnt_configs'])
-  if args.tag or args.tag_only:
-    LOGGER.info('Tagged configs: %u', len(counts['cnt_tagged_configs']))
-
+  counts = import_cfgs(args, dbt, logger)
 
-if __name__ == '__main__':
-  main()
+  logger.info('New configs added: %u', counts['cnt_configs'])
+  if args.import_configs.tag or args.import_configs.tag_only:
```

Comment:
Not a fan of the layered namespace changes in this file. Could we pass args.import_configs to run_import_configs as args?

Reply:
Subcommands or subparsers create layers, or levels of arguments. It's the only way we can make sense of so many argument groups. Its a concept we will have to get used to, unless you can come up with a better way to address the issue.

Reply:
import_configs is compartmentalized already, it has no knowledge of the other subcommands, so why should it be holding onto their namespaces?
I'm suggesting we constrict to the relevant namespace at input to the function defined here.